src/main/java/org/carlspring/cloud/storage/s3fs/S3Factory.java

```diff
@@ -0,0 +1,326 @@
+
+    public static final String PATH_STYLE_ACCESS = "s3fs_path_style_access";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(S3Factory.class);
+
+    private static final String DEFAULT_PROTOCOL = Protocol.HTTPS.toString();
+
+    private static final Region DEFAULT_REGION = Region.US_EAST_1;
+
+    /**
+     * Build a new Amazon S3 instance with the URI and the properties provided
+     *
+     * @param uri   URI mandatory
+     * @param props Properties with the credentials and others options
+     * @return {@link software.amazon.awssdk.services.s3.S3Client}
+     */
+    public S3Client getS3Client(final URI uri,
+                                final Properties props)
+    {
+        S3ClientBuilder builder = getS3ClientBuilder();
+
+        if (uri.getHost() != null)
+        {
+            final URI endpoint = getEndpointUri(uri.getHost(), uri.getPort(), props);
+            builder.endpointOverride(endpoint);
+        }
+
+        builder.region(getRegion(props))
+               .credentialsProvider(getCredentialsProvider(props))
+               .httpClient(getHttpClient(props))
+               .overrideConfiguration(getOverrideConfiguration(props))
+               .serviceConfiguration(getServiceConfiguration(props));
+
+        return createS3Client(builder);
+    }
+
+    private URI getEndpointUri(final String host,
+                               final int port,
+                               final Properties props)
+    {
+        String endpointStr;
+        if (port != -1)
+        {
+            endpointStr = String.format("%s:%d", host, port);
+        }
+        else
+        {
+            endpointStr = host;
+        }
+
+        // Endpoint may not have an scheme. E.g: 127.0.0.1:12345
+        if (!endpointStr.contains("://"))
+        {
+            final String protocol = getProtocol(props);
+            endpointStr = String.format("%s://%s", protocol, endpointStr);
+        }
+
+        return URI.create(endpointStr);
+    }
+
+    protected S3ClientBuilder getS3ClientBuilder()
+    {
+        return S3Client.builder();
+    }
+
+    private String getProtocol(final Properties props)
+    {
+        if (props.getProperty(PROTOCOL) != null)
+        {
+            return Protocol.fromValue(props.getProperty(PROTOCOL)).toString();
+        }
+
+        return DEFAULT_PROTOCOL;
+    }
+
+    private Region getRegion(final Properties props)
+    {
+        if (props.getProperty(REGION) != null)
+        {
+            return Region.of(props.getProperty(REGION));
+        }
+
+        try
+        {
+            return new DefaultAwsRegionProviderChain().getRegion();
+        }
+        catch (final SdkClientException e)
+        {
+            LOGGER.warn("Unable to load region from any of the providers in the chain");
+        }
+
+        return DEFAULT_REGION;
+    }
+
+    protected SdkHttpClient getHttpClient(final Properties props)
```

Comment:
Thanks for raising this! I believe it would be best to handle this in a follow-up task. @ptirador : Would you like to create one and back link it to here? :)